### PR TITLE
[FTAMS102020] Roy-Arne Brussel

### PR DESCRIPTION
**Pull request by Roybrussel**

### CODE REVIEW

File: src/movies.js [hunk 1]
```diff
@@ -1,16 +1,89 @@
...
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+const howManyMovies = listOfMovies => {
+    let stevenMovies = listOfMovies.filter(list => list['director'] === 'Steven Spielberg');  
+    let solution = stevenMovies.filter (dramas => dramas['genre'].includes('Drama')); 
+    return solution.length;}
+
```
Comment (LloydC) [inline src/movies.js:16]: Good work Roy!

Here's an alternative way to write function howManyMovies

`function howManyMovies(someMovies) {

return someMovies.filter(
eachMovie =>
eachMovie.director === 'Steven Spielberg' &&
eachMovie.genre.includes('Drama')
).length;
}`

File: src/movies.js [hunk 2]
```diff
@@ -1,16 +1,89 @@
...
+    
+    var total = 0;
+     
+    let grandTotal = sumRatings.forEach(function(number){
+      total += number;
+    })
+
+    return Number ((avgRating = total / sumRatings.length).toFixed(2));
+};
+
```
Comment (LloydC) [inline src/movies.js:61]: You can replace lines 51-59 with `return ratesAverage(arr)`. Also I think lines 47-49 would be better placed at the top before line 44 (it makes more sense to check the length of the array first then if it passes the condition the filtering can start)

File: src/movies.js [hunk 3]
```diff
@@ -1,16 +1,89 @@
...
+    let titlesList = arr.map(function (t) {
+      return t.title;
+    });
+    titlesList.sort();
+    if (titlesList.length < 20) {
+      return titlesList;
+    }
+    return titlesList.slice(0, 20);
+  }
+
```
Comment (LloydC) [inline src/movies.js:86]: Nice, I just want to show you a solution where everything is chained together

`function orderAlphabetically(lotsOfMovies) {

return [...lotsOfMovies]
.sort((a, b) => {
if (a.title > b.title) {
return 1;
} else if (a.title < b.title) {
return -1;
} else {
return 0;
}
})
.map(eachMovie => eachMovie.title)
.slice(0, 20);
}`

File: src/movies.js [hunk 4]
```diff
@@ -1,16 +1,89 @@
...
+};
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+const howManyMovies = listOfMovies => {
+    let stevenMovies = listOfMovies.filter(list => list['director'] === 'Steven Spielberg');  
+    let solution = stevenMovies.filter (dramas => dramas['genre'].includes('Drama')); 
+    return solution.length;}
```
Comment (wax-bear) [inline src/movies.js:15]: I would place the curly brace on a new line.

File: src/movies.js [hunk 5]
```diff
@@ -1,16 +1,89 @@
...
+    return listOfDirectors
+};
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+const howManyMovies = listOfMovies => {
+    let stevenMovies = listOfMovies.filter(list => list['director'] === 'Steven Spielberg');  
+    let solution = stevenMovies.filter (dramas => dramas['genre'].includes('Drama'));
```
Comment (wax-bear) [inline src/movies.js:14]: You might be interested in something called "method chaining". 

```
const howManyMovies = listOfMovies => {
    let spielbergDramaMovies = listOfMovies
      .filter(list => list['director'] === 'Steven Spielberg') 
      .filter (dramas => dramas['genre'].includes('Drama')); 

    return spielbergDramaMovies.length;
}
```

File: src/movies.js [hunk 6]
```diff
@@ -1,16 +1,89 @@
...
+  return Number((totalRate / arr.length).toFixed(2));
+}
+ratesAverage(ratingsOnly);
+
+// Please note that Jasmine still gives a red dot for not returning a new array
+// even though I am.
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(array) {
```
Comment (wax-bear) [inline src/movies.js:43]: Perhaps think about how you can optimize this function (less lines/methods).

File: src/movies.js [hunk 7]
```diff
@@ -1,16 +1,89 @@
...
+    let grandTotal = sumRatings.forEach(function(number){
+      total += number;
+    })
+
+    return Number ((avgRating = total / sumRatings.length).toFixed(2));
+};
+
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
 
+const orderByYear = moviesArray => {
```
Comment (wax-bear) [inline src/movies.js:64]: Good stuff!

**Review by wax-bear — COMMENTED** [review]
Great stuff Roy-Arne!

A few test cases in getAllDirectors and orderByYear are not working yet.
Please check out my comments for potential improvements.

**Pull request closed without merging**